ui/app/AppLayouts/Wallet/services/dapps/DAppsListProvider.qml

```diff
@@ -15,7 +18,28 @@ QObject {
     required property DAppsStore store
     required property var supportedAccountsModel
 
-    readonly property alias dappsModel: d.dappsModel
+    property string selectedAddress: ""
+
+    readonly property SortFilterProxyModel dappsModel: SortFilterProxyModel {
+        objectName: "DAppsModelFiltered"
+        sourceModel: d.dappsModel.count ? d.dappsModel : null
```

Comment:
It was done because SFPM once its source model is bind, and roles initialized, will not "refresh" roles for the usecases of `dynamicRole: true` of ListModel.
Right now when app starts, walletconnect service will send data to `d.dappsModel` that has slightly different structure then the following data entries.

Reply:
That's something not clear for me. `dynamicRole: true` only allows various types for the same role.
So it's valid then:
```
append({name:"X"})
append({name:4})
```
but
```
append({name:"X"})
append({name:4, surname: "Y"})
```
will result with a model with only `name` role. `surname` won't be added in a magic way.

And `SFPM` doesn't care about role types at all. So I wonder why it's needed here :)

Reply:
It looks like right now dApps are stored in local cache in slightly different format than what is actually needed to properly filter them by connected address. And when locally stored dApps are overwritten by up-to-date data from wallet connect, `dappsModel` is cleared first. `dynamicRoles: true` seems to be covering that case.
Anyways, based on our discussions with you and then with Stefan, I will adjust that data and remove both `dynamicRoles` and that binding in "DAppsModelFiltered" SFML.